File: metadata-consumer/src/main/java/com/chord/nft/metadata/model/TokenMetadata.java
package com.chord.nft.metadata.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.bson.Document;
import org.json.JSONObject;

import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TokenMetadata implements Serializable {
    private Document metadata;
    private String tokenId;
    private String address;
    private String assetCDNUrl;

    @Override
    public String toString() {
        return "TokenMetadata{" +
                "metadata=" + metadata +
                ", tokenId='" + tokenId + '\'' +
                ", address='" + address + '\'' +
                ", assetCDNUrl='" + assetCDNUrl + '\'' +
                '}';
    }
}

File: apis/packages/commons/src/models/token-metadata.ts
import mongoose, { Schema, Document } from "mongoose";

const metadataSchema: Schema = new Schema({
    fullname: {
        type: String
    },
    address: {
        type: String
    },
    assetCDNUrl: {
        type: String
    },
    metadata: {
        type: Schema.Types.Mixed
    }
}, { strict: false, collection: 'tokenMetadata' });

const Metadata = mongoose.model("tokenMetadata", metadataSchema, 'tokenMetadata');
export default Metadata;